script.js

```diff
@@ -1,4 +1,569 @@
+    outputValue += `${dealerHand[counter].name} of ${dealerHand[counter].suit} ${dealerHand[counter].suitIcon} | `;
+  }
+
+  return outputValue;
+};
+
+// Create a helper function for dealer state
+var dealerState = function (input) {
+  var outputValue = dealerActions();
+  var dealerScore = dealerStatus.score;
+  outputValue += `<br><br>Here is the results of the rounds:<br>`;
+  // Count player scores, reset player hands
+  for (playerIndex = 0; playerIndex < numPlayers; playerIndex += 1) {
+    var playerHand = playerStatus[playerIndex].hand;
+    var playerName = playerStatus[playerIndex].name;
+    var playerBet = playerStatus[playerIndex].bet;
+    var playerScore = playerStatus[playerIndex].score;
+    var playerBlackjack = playerStatus.blackjack;
+
+    outputValue += `<b>${playerName}:</b> <br> `;
+    for (handIndex = 0; handIndex < playerHand.length; handIndex += 1) {
+      var handScore = playerScore[handIndex];
+      outputValue += `Hand ${handIndex + 1}: ${handScore}; <br> `;
+
+      if (playerBlackjack == true) {
+        playerStatus[playerIndex].credits += playerBet * 3;
+        outputValue += `With a bet of $${playerBet}, you have won $${
+          playerBet * 2
+        }. You have a remaining credit of $${
+          playerStatus[playerIndex].credits
+        }<br>`;
+      } else if (handScore > dealerScore) {
+        playerStatus[playerIndex].credits += playerBet * 2;
+        outputValue += `With a bet of $${playerBet}, you have won $${playerBet}. You have a remaining credit of $${playerStatus[playerIndex].credits}<br>`;
+      } else if (handScore == dealerScore) {
+        playerStatus[playerIndex].credits += playerBet;
+        outputValue += `You have drawn with the dealer. You have a remaining credit of $${playerStatus[playerIndex].credits}<br>`;
+      } else {
+        outputValue += `You have lost this hand. You have a remaining credit of $${playerStatus[playerIndex].credits}<br>`;
+      }
+    }
+    playerStatus[playerIndex].score = [0];
+    playerStatus[playerIndex].bet = 0;
+    playerStatus[playerIndex].hand = [];
+    playerStatus[playerIndex].blackjack = false;
+    outputValue += "-------------------------------------<br><br>";
+  }
+  // Reset Banker hands
+  dealerStatus.hand = [];
+  dealerStatus.score = 0;
+  currentPlayer = 1;
+  gameState = 2;
+
+  outputValue += `We will now return to the betting phase to play another game.`;
+  return outputValue;
+};
+
+// Main Function
+// Note: To add transition from betting to player 1's turn.
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  if (gameState == 0) {
+    return setNumPlayer(input);
+  } else if (gameState == 1) {
+    return setName(input);
+  } else if (gameState == 2) {
+    return setBet(input);
+  } else if (gameState == 3) {
+    return playerState(input);
+  } else if (gameState == 4) {
+    return dealerState(input);
+  }
+};
```

Comment:
good job using game states to switch into different modes but the naming of the game states could be clearer. It is not very clear to me what 1-4 represents

Reply:
same comment as above regarding the use of too many ifs and else ifs